client/src/js/game-channel/game-channel.js

```diff
@@ -608,6 +631,28 @@ export class GameChannel {
     this.gameRound.isCompleted = true;
     this.gameRound.userInAction = false;
     this.saveStateToLocalStorage();
+
+    let winnerName = undefined;
+    switch (winner) {
+      case this.channelConfig.initiatorId:
+        winnerName = 'Bot';
+        break;
+      case this.channelConfig.responderId:
+        winnerName = 'User';
+        break;
+    }
+
+    const description = winnerName ? `${winnerName} wins!` : "It's a draw!";
+
+    const transactionLog = {
+      id: `${this.gameRound.index}`,
+      description,
+      signed: SignatureTypes.confirmed,
+      onChain: false,
+      timestamp: Date.now(),
+    };
+    addInfoLog(transactionLog, this.gameRound.index);
+
```

Comment:
Considering the fact that our middleware is executed whenever the game state changes, would it be possible to execute `dom-manipulation` functionalities inside middleware like we do with the rest dom manipulations? I think that approach could make code less bloated

Reply:
This is a suggestion - If it does not sound like a good idea, feel free to merge

@martinkaintas